src/main/webapp/app/exercises/shared/statistics/doughnut-chart.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { DoughnutChartType } from 'app/course/manage/detail/course-detail.component';
+import { CourseStatisticsDataSet } from 'app/overview/course-statistics/course-statistics.component';
+import { ChartType } from 'chart.js';
+import { round } from 'app/shared/util/utils';
+import { ExerciseType } from 'app/entities/exercise.model';
+
+@Component({
+    selector: 'jhi-doughnut-chart',
+    templateUrl: './doughnut-chart.component.html',
+    styleUrls: ['./doughnut-chart.component.scss'],
+})
+export class DoughnutChartComponent implements OnChanges, OnInit {
+    @Input() courseId: number;
+    @Input() contentType: DoughnutChartType;
+    @Input() exerciseId: number;
+    @Input() exerciseType: ExerciseType;
+    @Input() currentPercentage: number | undefined;
+    @Input() currentAbsolute: number | undefined;
+    @Input() currentMax: number | undefined;
+
+    receivedStats = false;
+    doughnutChartTitle: string;
+    stats: number[];
+    titleLink: string[] | undefined;
+
+    constructor(private router: Router) {}
+
+    // Chart.js data
+    doughnutChartType: ChartType = 'doughnut';
+    doughnutChartColors: any[] = ['limegreen', 'red'];
+    doughnutChartLabels: string[] = ['Done', 'Not Done'];
+    totalScoreOptions: object = {
+        cutoutPercentage: 75,
+        scaleShowVerticalLines: false,
+        responsive: false,
+        tooltips: {
+            backgroundColor: 'rgba(0, 0, 0, 1)',
+            callbacks: {
+                label(tooltipItem: any, data: any) {
+                    const value = data['datasets'][0]['data'][tooltipItem['index']];
```

Comment:
This just looks like a ticking time bomb 😄 One undefined and boom 💥

Reply:
Agree on the 💥 , I took it from the internet and it worked until now, but I added an additional check for safety reasons now :) 0c371b3